internal/announcements/announce.go

```diff
@@ -0,0 +1,94 @@
+package announcements
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Event struct {
+	Type string
+	Data map[string]interface{}
+}
+
+func NewEvent(eventType string, initialData ...map[string]interface{}) *Event {
+	var data map[string]interface{}
+	if len(initialData) > 0 {
+		data = initialData[0] // Use the first map provided if any.
+	} else {
+		data = make(map[string]interface{})
+	}
+
+	return &Event{
+		Type: eventType,
+		Data: data,
+	}
+}
+
+func (e *Event) Set(key string, value interface{}) {
+	e.Data[key] = value
+}
+
+func (e *Event) Get(key string) (interface{}, error) {
+	value, exists := e.Data[key]
+	if !exists {
+		return nil, fmt.Errorf("key '%s' does not exist in event of type '%s'", key, e.Type)
+	}
+	return value, nil
+}
+
+type Announcement interface {
+	Announce(event *Event)
+}
+
+type NopAnnouncement struct{}
```

Comment:
I hate to be that guy but should this be `NoopAnnouncement` (I know I know, different amount of letters)